anchore_engine/services/policy_engine/engine/policy/gates/vulnerabilities.py

```diff
@@ -218,10 +219,18 @@ def evaluate(self, image_obj, context):
 
                                     if is_fix_available and fix_timeallowed is not None:
                                         if fix_available_in:
-                                            if calendar.timegm(vulnerability_cpe.created_at.timetuple()) > fix_timeallowed:
+                                            fix_date = vulnerability_cpe.created_at
+
+                                            # VulnDB entries might have a solution_date which is more accurate than the CPE created_at field
+                                            if vulnerability_cpe.parent.vuln_metadata:
```

Comment:
This is accessing `vuln_metadata` attribute under the assumption that `parent` is an instance of `VulnDBMetadata`. It will fail for other types such as nvd cpes with parent is `NvdV2Metadata` that are schlepped through the same gate logic. Checks and additional error handling are necessary here

Reply:
Missed this comment. Good point as well, so we should test both if `vuln_metadata` exists and if `solution_date` is a field.